### PR TITLE
add alternative onExcessiveCPUUsage method

### PR DESCRIPTION
**Pull request by magne4000**
New method `onExcessiveCPUUsageSimple` that uses https://github.com/simonepri/pidtree and https://github.com/soyuka/pidusage to calculate CPU usage

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,105 +1,164 @@
...
   /**Number of samples to consider */
-  samplesCount?: number
+  samplesCount?: number;
   /**CPU usage percent minimum to consider a sample exceeds CPU usage */
-  percentCPUUsageThreshold?: number
+  percentCPUUsageThreshold?: number;
+}
+
+/**
+ * Will emit an array `PidUsage` when a process exceeds the
```
Comment (alexstrat) [inline src/index.ts]: Worth mentioning it monitor the CPU usage of the give process and its children. I have hard time figuring it out.

File: src/index.ts [hunk 2]
```diff
@@ -1,105 +1,164 @@
...
+ * `options.percentCPUUsageThreshold` on more than `options.samplesCount`
+ * samples.
+ *
+ * Default `options.samplesCount` = 10
+ * Default `options.percentCPUUsageThreshold` = 80
+ *
+ * @param pid the pid of the main process
```
Comment (alexstrat) [inline src/index.ts]: Of any process actually

Comment (alexstrat) [reply src/index.ts]: To keep consistency with the original `onExtendedProcessMetrics` I'd rather take `app` as argument and extract the main process id from it.

**Comment by alexstrat** [comment]
Update README

File: src/index.ts [hunk 3]
```diff
@@ -1,105 +1,164 @@
...
- * options.samplingInterval = 1000 (1s) by default 
- * @param app 
- * @param options 
+ *
+ * options.samplingInterval = 1000 (1s) by default
+ * @param app
+ * @param options
  */
-export const onProcessMetrics = (app: Electron.App, options: onProcessMetricsOptions) => {
+export const onProcessMetrics = (pid: number, options: onProcessMetricsOptions): Observable<PidUsage[]> => {
```
Comment (alexstrat) [inline src/index.ts]: I'd prefer you don't include breaking changes to the external API. Either add retro-compatible changes to this API or add an new API.

File: src/index.ts (hunk 1 above)
Comment (alexstrat) [reply src/index.ts]: Worth mentioning also explains the difference with the original `onExcessiveCPUUsage`.

File: src/index.ts (hunk 2 above)
Comment (magne4000) [reply src/index.ts]: We do not really have a way to get a pid from the `app` object

**Comment by magne4000** [comment]
@alexstrat comments taken into account. When you consider this is OK, you can release it as a new version.

**Comment by alexstrat** [comment]
Published as `electron-process-reporter@1.3.0`

**Pull request merged**